### DISCUSSION TITLE
making an auth token config

### DISCUSSION THREAD

**Discussion by DannyDarth** [streamlink/streamlink-twitch-gui#939] [Q&A] [answered]
If i am reading right. To make an auth token config for twtich. i need to add it to %APPDATA%\streamlink\config.pluginname,
naming the new config to config.twitch? Then adding the line "--twitch-access-token-param KEY=(twitch token)".

Is this how i am suppose to do it with it using my own auth token?

**Comment by bastimeyer** [comment, marked as answer]
Everything is explained very clearly in Streamlink's docs.
https://streamlink.github.io/cli/plugins/twitch.html#authentication

On Windows, the config file is located at `%APPDATA%\streamlink`, so either create/update your `config` file or add `config.twitch` and then add the necessary parameters WITHOUT leading dashes, as explained clearly in the config docs:
https://streamlink.github.io/cli/config.html#syntax

> `--twitch-access-token-param KEY=(twitch token)`

How did you come up with this? This is mentioned nowhere in the docs.

You are supposed to add an OAuth token like this, which is **once again** [clearly mentioned in the docs](https://streamlink.github.io/cli/plugins/twitch.html#authentication):
```
twitch-api-header=Authorization=OAuth YOUR_TOKEN_HERE
```

**Comment by DannyDarth** [comment]
So i did that without the -- just now and this is what pops up when i try to open a stream

https://i.imgur.com/rdHmOm9.png

Referenced the command line from here 
https://streamlink.github.io/latest/cli.html#cmdoption-config

EDIT: Followed what you just posted. with the command line and its working. Sorry for the hassle i miss understood what was posted.

**Reply by bastimeyer** [reply]
I can't help people who refuse to read, sorry. This is not difficult at all, and as I already said several times after the recent ad-changes on Twitch, I won't spend any more time on individual people who are having trouble understanding any of this, especially when everything is already documented as well and as comprehensible as possible. I can't help everyone, and the vast majority of people can make this work without any issues. This is not worth my time, and I have already wasted several ten thousands of hours on this project. Be glad that you already received your answer. Good luck.

**Comment by fuguesoft** [comment]
Am I doin it rite?

![imagen](https://github.com/user-attachments/assets/44a90600-e830-4fa1-8047-f64013c74f3c)

**Reply by fuguesoft** [reply]
I only have the one line `twitch-api-header=Authorization=OAuth TOKEN` in that file and I'm using the python build on Debian 12.

I made the directory and file myself but don't know how to tell if streamlink is actually looking at it.

the only way to test is to watch a stream I'm subbed in to see if there are ads and the stream pauses but that will take an hour

**Reply by fuguesoft** [reply]
I guess my question is, "is there a way to make streamlink and by extension streamlink-twitch-gui confirm which file it's using?"
